channeldb/invoices.go

```diff
@@ -385,6 +386,10 @@ type InvoiceHTLC struct {
 	// payment for internal processing.
 	AMP *record.AMP
 
+	// Preimage is an HTLC-level preimage provided by the sender as part of
+	// an AMP payment. This value will be nil for legacy or MPP payments.
+	Preimage *lntypes.Preimage
```

Comment:
Exposing `Preimage` and `AMP` on the rpc would be useful for testing this PR. And is probably needed anyway.

Reply:
added, w/ itest assertions on the returned fields